system_tests/bigquery.py

```diff
@@ -329,13 +333,14 @@ def test_load_table_from_storage_then_dump_table(self):
 
         job.begin()
 
-        counter = 9  # Allow for 90 seconds of lag.
+        def _job_done(instance):
+            return instance.state in ('DONE', 'done')
 
-        while job.state not in ('DONE', 'done') and counter > 0:
-            counter -= 1
-            job.reload()
-            if job.state not in ('DONE', 'done'):
-                time.sleep(10)
+        # Allow for 90 seconds of "warm up" before rows visible.  See:
+        # https://cloud.google.com/bigquery/streaming-data-into-bigquery#dataavailability
+        # 7 tries -> 2**7 + 2**6 + ... 1 = 127 seconds.
+        retry = RetryInstanceState(_job_done, max_tries=7)
```

Comment:
It seems that `RetryResult` could be used without having to also define `RetryInstanceState`

```python
def _job_reload():
    job.reload()
    return job

retry = RetryResult(_job_done, max_tries=7)
retry(_job_reload)()
```